### ISSUE TITLE
to_feather() got an unexpected argument 'compression'

### ISSUE THREAD

**Issue by ccunningham101** [iaindillingham/cohort-joiner#25]
v.0.0.9 of the cohort-joiner is getting the error "to_feather() got an unexpected argument 'compression'" with python docker image sha256:a0bfa308bd8a7deed52022deb71a6c9708bf8559713a73fb03e36bb5bf40c5a1 
Error seen with [hypertension ](https://github.com/opensafely/hypertension-sro/runs/5812774900?check_suite_focus=true#step:2:8854) project
Related to [this ](https://github.com/opensafely-core/cohort-extractor/commit/72816cadcf487d1f7b29f9489aa87769dc03a5f0) update?

**Comment by iaindillingham** [comment]
tl;dr dependencies 😬 

cohort-joiner is run within a container crated from our python image, which is created from our python-docker repo. Currently, this container's Python environment includes `opensafely-cohort-extractor==1.63.1` and `pandas==1.0.1`; that is, these packages and versions are given by our python-docker repo's _requirements.txt_.

However, cohort-extractor v1.63.1 requires `pandas==1.4.1` (see [_requirements.prod.txt_](https://github.com/opensafely-core/cohort-extractor/blob/v1.63.1/requirements.prod.txt)). Somewhere between Pandas v1.0.1 and v1.4.1, `DataFrame.to_feather` started to accept the `compression` keyword argument. We started to use this keyword argument in [this commit](https://github.com/opensafely-core/cohort-extractor/commit/72816cadcf487d1f7b29f9489aa87769dc03a5f0), which was released in cohort-extractor v1.63.1.

Consequently, when `cohortextractor.pandas_utils.dataframe_to_file` is called within a container created from our cohort-extractor image, its Python environment includes Pandas v1.4.1. When it is called within a container created from our python image, its Python environment includes Pandas v1.0.1. In the former case, passing a feather file works; in the latter case, passing a feather file doesn't work.

#13 replaced copy-and-paste code from cohort-extractor with a call to cohort-extractor itself. Whilst we could revert the commits in this PR, we'd still have an implicit dependency between cohort-joiner and cohort-extractor.
